client/src/components/Format/Format.component.tsx

```diff
@@ -0,0 +1,107 @@
+import React from 'react'
+
+// ---| core |---
+import { cn } from 'tools'
+
+// ---| pages |---
+// ---| screens |---
+// ---| components |---
+import { Text, TextProps } from 'components/Text'
+
+// ---| self |---
+import css from './Format.module.scss'
+import {
+  DEFAULT_SIZE_OPTIONS, SIZE_OPTIONS, DEFAULT_WEIGHT_OPTIONS, WEIGHT_OPTIONS,
+  toString, toNumber, toUnit, toDate, capitalize, getMaxOptions,
+} from './Format.tool'
+
+export type FormatValue = string | number | Date | undefined | null
+
+const STRING_FORMAT_MAP = {
+  'uppercase': (v: FormatValue) => toString(v).toUpperCase(), // ABC DEF
+  'lowercase': (v: FormatValue) => toString(v), // abc def
+  'capitalize': (v: FormatValue) => {
+    const whitespace = ' '
+
+    return toString(v).split(whitespace).map(capitalize).join(whitespace)
+  }, // Abc Def
+  'title': (v: FormatValue) => capitalize(toString(v)), // Abc def
+}
+
+const NUMBER_FORMAT_MAP = {
+  // 123,456,789.98
+  'number': (v: FormatValue) => toNumber(v, 'decimal', 2),
+
+  // 123,456,789
+  'amount': (v: FormatValue) => toNumber(v, 'decimal', 0),
+
+  // 123,456,789.98 %
+  'percent': (v: FormatValue) => toNumber(v, 'percent', 2),
+
+  // 123.46 M$  - K, M, B
+  'currency': (v: FormatValue) => toNumber(v, 'currency', 2, 'USD', 'compact'),
+
+  // 123.46 TB  - B, KB, MB, GB, TB
+  'size': (v: FormatValue) => {
+    const options = getMaxOptions(v, SIZE_OPTIONS, DEFAULT_SIZE_OPTIONS)
+    const intlValue = toUnit(v, SIZE_OPTIONS, DEFAULT_SIZE_OPTIONS).format(Number(v/options.value))
+
+    return isNaN(intlValue) ? `0 ${options.postfix}` : `${intlValue} ${options.postfix}`
+  },
+
+  // 123.46 T  - G, K, T,
+  'weight': (v: FormatValue) => {
+    const options = getMaxOptions(v, WEIGHT_OPTIONS, DEFAULT_WEIGHT_OPTIONS)
+    const intlValue = toUnit(v, WEIGHT_OPTIONS, DEFAULT_WEIGHT_OPTIONS).format(Number(v/options.value))
+
+    return isNaN(intlValue) ? `0 ${options.postfix}` : `${intlValue} ${options.postfix}`
+  },
+}
+
+const DATE_FORMAT_MAP = {
+  'day': (v: FormatValue) => toDate(v, {day: '2-digit'}),
```

Comment:
Can you add comment after string how it looks like? example
`'day': (v: FormatValue) => toDate(v, {day: '2-digit'}), // 13`